### PR TITLE
Parse stream as bytearray instead of using upfrom BitStream

### PR DESCRIPTION
**Pull request by pauloasherring**
Hello @slhck ,
First of all, thanks for this fine piece of software :)
Resources for h264 are not really widespread, so, I've used this as a learning tool.

The changes I am proposing are basically to, instead of parsing initial NALUs using the BitStream() type, start by parsing it as bytearray().

With three different h264 files, the output of the proposed class and the original class is almost exactly the same (two differences are listed below), but, performance-wise, the proposed approach resulted in roughly 100 times faster parsing in verbose mode.

Output differences:

* `print("Warning: 3-byte extra NAL unit start code found")` -> I am really not making difference on whether the could NALU is "3-bytes extra" or not;
* `print ("Parsing NALU %d" % nalcount)` -> this was removed, the parsing is happening fast enough, so I opted to remove it.

Should you have any considerations in regards to this, please, let me know.
Thanks again and kind regards.

### CODE REVIEW

**Comment by slhck** [comment]
Thank you for creating this pull request! I am definitely considering merging this. Will be reviewing it in a second.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 # Changelog
 
+## v0.7.1 (2021-11-12)
+
+* Changes to H26xParser, to make it more performatic. First extraction occurs using bytearrays instead of Bitstream. Performed benchmark resulted roughly in a reduction from 250s to 2.5s.
```
Comment (slhck) [inline CHANGELOG.md]: Please undo the changelog change. I will generate the changelog automatically when I make a new release.

File: README.md [hunk 2]
```diff
@@ -1,8 +1,12 @@
 # h26x-extractor
 
+## Why the fork?
+
+When I've started working on this, I disregarded whether the original author would or would not accept pull requests. So, this was why.
+
```
Comment (slhck) [inline README.md]: Please undo this part so I can merge into my repo.

File: h26x_extractor/__main__.py [hunk 3]
```diff
@@ -9,21 +9,47 @@
...
-    main()
+    if args["--verbose"] == True:
+      timeit(main)()
+    else:
+      main()
+    if args["--verbose"] == True:
+      print('Verbose is true')
+    else:
+      print('Verbose is false')
+
```
Comment (slhck) [inline h26x_extractor/__main__.py]: This looks a bit clunky to me. Note that `if x == True` is not pythonic. Simply use `if x`.

```python
start = timeit.default_timer()
main()
stop = timeit.default_timer()
if args["--verbose"]:
    print('Time: ', stop - start)  
```

File: h26x-extractor-localrun.py [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+if __name__ == "__main__":
+    if args["--verbose"] == True:
+      timeit(main)()
+    else:
+      main()
+    if args["--verbose"] == True:
+      print('Verbose is true')
+    else:
+      print('Verbose is false')
```
Comment (slhck) [inline h26x-extractor-localrun.py]: What is this file for? You can run the CLI part with `python3 -m h26x_extractor`, so this file is not really needed. Please remove it from the commit.

**Review by slhck — CHANGES_REQUESTED** [review]

File: h26x-extractor-localrun.py (hunk 4 above)
Comment (pauloasherring) [reply h26x-extractor-localrun.py]: _Doing `python3 -m h26x_extractor` requires module install, no?_
> Nevermind this. Just tested and it works! Thanks for the hint!

I've used this for local testing, without installation or any other subterfuge.
But, anyways, I'll remove it.

File: README.md (hunk 2 above)
Comment (pauloasherring) [reply README.md]: Rolled back.

File: CHANGELOG.md (hunk 1 above)
Comment (pauloasherring) [reply CHANGELOG.md]: Rolled back.

File: h26x-extractor-localrun.py (hunk 4 above)
Comment (slhck) [reply h26x-extractor-localrun.py]: No, you don't need to install the module. You can just directly run it from the source code, assuming you have the dependencies (docopt, ...) installed too.

File: h26x_extractor/__main__.py (hunk 3 above)
Comment (pauloasherring) [reply h26x_extractor/__main__.py]: Sorry, python is not really my first language 😞 
Changed it.

File: h26x-extractor-localrun.py (hunk 4 above)
Comment (pauloasherring) [reply h26x-extractor-localrun.py]: Thanks for the hint! It indeed worked.

**Comment by pauloasherring** [comment]
Requested changes have been done.

**Comment by slhck** [comment]
Thanks for making the changes! I noticed there were problems with Windows CRLF line endings in the committed files. If you are on Windows and use Git, please set `git config --global core.autocrlf true`.

I also reinstated the function to pass a raw bitstream as a hex-formatted string for testing purposes.

I will merge the changes manually from a separate branch.

Thanks for your contributions; they are much appreciated!

**Comment by slhck** [comment]
PS: Doing a simple test this brings down the analysis time from ~6.8 seconds to 140 ms 🎉

**Comment by slhck** [comment]
PPS: I noticed you were pushing your changes to the master branch on your own fork. In case you plan doing PRs on other people's repositories, it is better to use a dedicated branch to make your changes. For instance, after you fork my repo, you can do:

```
git switch -c my-new-feature
# do stuff
git add .
git commit -m "my changes"
git push origin my-new-feature
```

This way, your changes can be tracked on a dedicated branch and I can also modify them directly, without changing your fork's master branch.

**Comment by pauloasherring** [comment]
Alright! I've never used the fork feature, first timer here. The projects I work, forking usually is disabled :(
But, still, glad to contribute :)
Again, thanks for the fine piece of software!

**Pull request closed without merging**